Adds functionality to choose the pool for Ice Trap appearance

dotzo opened the pull request:
Possible looks_like_item values pull first from the existing item pool.  If there's not enough stuff it'll generate on its own from the table / hardcoded lists.  

Solves the problem of bait freestanding / big chests without outright removing icetraps altogether.

r0bd0g commented:
There's one weird case, the chest minigame, where the item you see and the chest size need to agree. Just wanted to mention that, to make sure there's nothing special you have to do for that?

dotzo commented:
I believe CSMC runs off the item attribute looks_like_item.  If that's not the case, then none of CSMC will necessarily work ith this.  Asked in discord dev channels but no answer.  I'm good updating if there's issue.

dotzo commented:
Cleaned up FIll a little bit, removed some duplicate imports and unused imports.  


I investigated the CSMC thing and the code is only ever referencing the item_id for chest size/color.  Ice traps have a chest type of 0, meaning they should be big.  The override_t.value struct has a parameter for looks_like which is never used.  I believe that should be what's used in the chest.c if the item has one when referencing type/color.  

I'm not able to compile the C so any changes won't take effect.
